frontend/src/components/SelectDocuments.js

```diff
@@ -0,0 +1,138 @@
+import React, { Component } from 'react'
+import {Selector} from './Selector'
+import { getAllDocumentClasses } from '../utils/ApiWrapper'
+import { bindActionCreators } from 'redux'
+import { connect } from 'react-redux'
+import DatePicker from "react-datepicker";
+
+
+const mapStateToProps = state => ({
+  isPM: state.user.isPM
+})
+
+const mapDispatchToProps = dispatch => {
+  return bindActionCreators(
+    {
+      //put actions here
+    },
+    dispatch
+  )
+}
+class SelectDocumentsPage extends React.Component {
+  constructor(props) {
+    super(props)
+    var today = new Date()
+    // var dd = String(today.getDate()).padStart(2, '0');
+    // var mm = String(today.getMonth() + 1).padStart(2, '0'); //January is 0!
+    // var yyyy = today.getFullYear();
+    // today = mm + '/' + dd + '/' + yyyy
+    this.state = {
+        docClass: {},
+        filtered: {},
+        DueDate: today,
+        query: ''
+    }
+  }
+
+  async componentDidMount() {
+    let documents = await getAllDocumentClasses()
+    this.setState(this.updateDocumentClasses(documents))
+    
+  }
+
+  updateDocumentClasses(res) {
+    if (res) {
+        let docList = {}
+        for (var i of res) {
+            docList[i] = 'Available'
+        }
+        return {
+            docClass: docList,
+            filtered: docList
+        }
+    } else {
+        return {
+            docClass: {}
+        }
+    }
+  }
+
+  handleQueryChange = (event) => {
+    let newState = this.state
+    let query = event.target.value.toLowerCase()
+    newState['query'] = query
+    newState['filtered'] = {}
+    if (query == "") {
+      newState['filtered'] = this.state.docClass
+    } else {
+      Object.keys(this.state.docClass).map(key => {
+        if (key.toLowerCase().includes(query)) {
+          newState['filtered'][key] = this.state.docClass[key]; 
+        }
+      })
+    }
+    this.setState(newState);
+    console.log(this.state)
+  }
```

Comment:
Some comments would be beneficial in this function and even in the function below this. Either that or I'm super tired, but it never hurts to have comments in either case.

Reply:
NANI